components/configurator/view/SaveConfiguratorView.js

```diff
@@ -43,7 +49,56 @@ const SaveConfiguratorView = () => {
     })
   }, [setStatusIndicator, step])
 
+  useEffect(() => {
+    if (!recaptchaValidated && recaptchaRef.current) {
+      console.log('ReCAPTCHA initializing') // eslint-disable-line no-console
+      recaptchaRef.current.execute()
+    }
+  }, [recaptchaValidated, recaptchaRef])
+
+  const onReCAPTCHAChange = async code => {
```

Comment:
Not a big deal but I think it would be cleaner to encapsulate as much of this recaptcha logic in the `ConfiguratorRecaptcha` component as possible.

I believe you can expose additional callbacks in the component that will allow you to simplify its consumption here in `SaveConfiguratorView`.

Something like this should do it:
```
const handleReCAPTCHASuccess = () => {
  setRecaptchaValidated(data.success)
  setIsHuman(data.success)
  setStep(STEPS.READY)
}

const handleReCAPTCHAFail = () => {
  setErrorMessage('error: are you human?')
  playSound('button_cancel.wav')
  setStep(STEPS.ERROR)
}

/* ... */

<ConfiguratorRecaptcha
  onReCAPTCHASuccess={handleReCAPTCHASuccess}
  onReCAPTCHAFail={handleReCAPTCHAFail}
/>

```
This should allow you to move a few items to `ConfiguratorRecaptcha`:
- The API call to `/api/verify-human`
- The `recaptchRef`
- Recaptcha exception handling and reset logic

I surely misspelled "recaptcha" at least once in this comment 😄

Reply:
Great call, @seanconnollydev on it!

Reply:
All set! Ready for one more review!